### PR TITLE
[Howto guides] Install wechaty

### LINKED ISSUE #896: Update and enhance how-to-guide section
This issue is to track :
* What has been done and 
* What we are on currently working :)

## Team members
Abhishek Jaiswal - @abhishek-iiit 
Vasvi Sood  - @vasvi-sood 

# References 
- [x] **Readme(to github section)** 
- [x] **Overview** 
- [x] **Add events to the bot**
    - [x] Add events and customise a bot
- [x] **Add more functionality to your bot**
    - [x] Dealing with messages
    - [x] Manage Contacts
    - [x] Manage Rooms
    - [x] Send and Accept Friend Requests
    - [x] Sending and Receiving files
- [x] **Deploy on IM platfrom**
    - [x] Deploy in Wechat
    - [x] Deploy in Whatsapp
    - [x] Deploy in Lark
    - [x] Deploy in Gitter
    - [x] Deploy in Wechat Official Account
    - [x] Deploy in PadLocal
    - [x] Deploy in WeChat4U
- [x] **Overview with containers**
    - [x] Deploy with Heroku
    - [x] Deploy with Docker

Please link pull request to this section for the easy access.

### PR DESCRIPTION
**Pull request by vasvi-sood**
The starter.md file contains a step by step guide on how to run the wechaty bot with the most famous messaging apps- Wechat and Whatsaap

### CODE REVIEW

**Comment by proudofsimin** [comment]
I saw quite a lot of files but did not see the starter.md. Could you screenshot where we should review? Thank you.

**Review by abhishek-iiit — COMMENTED** [review]
Please review for the same

**Comment by Rohitesh-Kumar-Jain** [comment]
> Could you screenshot where we should review? Thank you.

Yes pls add screenshots, ask for help if you need help in adding screenshots of the preview : )

**Comment by vasvi-sood** [comment]
> > Could you screenshot where we should review? Thank you.
> 
> Yes pls add screenshots, ask for help if you need help in adding screenshots of the preview : )

Screenshot
This is the starter.md file. This can be reviewed at  docusaurus/docs/howto/starter.md.

**Comment by Rohitesh-Kumar-Jain** [comment]
Hey @vasvi-sood and @abhishek-iiit, I would be glad if you may keep your commit history clean next time you raise a PR.

By clean commit history and a good screenshot, I mean something like this : #928 

Consider squashing commits, and try keeping the commit history easy to review.

**Comment by Rohitesh-Kumar-Jain** [comment]
This PR neither have screenshot(s) nor clean commit history making it hard to review 😅 

I am confused, from where should I start reviewing 51 commits ?

**Comment by vasvi-sood** [comment]
We'll add a related screenshot. We had some issue in pushing the branch to GitHub due to an error in package.json file. We had to make various commits for that. We'll take care of it from next time.

**Comment by vasvi-sood** [comment]
Screenshot from 2021-06-11 12-39-31
Screenshot from 2021-06-11 12-39-55
Screenshot from 2021-06-11 12-40-17

**Comment by vasvi-sood** [comment]
This is how the starter.md file looks. Kindly review it @proudofsimin  and @Rohitesh-Kumar-Jain

File: docusaurus/docs/howto/starter.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+title: 'Getting started with the bot on Wechat or Whatsaap'
```
Comment (proudofsimin) [inline docusaurus/docs/howto/starter.md:2]: correct the typo: whatsapp

File: docusaurus/docs/howto/starter.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+This will be a step by step guide on how to run the bot on most popular messaging apps, Whatsapp and Wechat:
```
Comment (proudofsimin) [inline docusaurus/docs/howto/starter.md:5]: How do you think if we change to:
In this section, you will learn how to run the bot on the popular messaging apps, including WhatsApp and Wechat.

File: docusaurus/docs/howto/starter.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+:::tip
+
+If you do not wish to run the bot locally, you can also use use Gitpod  or Google Cloud Shell. The steps can be found [here](../getting-started/quick-start)
```
Comment (proudofsimin) [inline docusaurus/docs/howto/starter.md:9]: please remove the duplicated *use*

**Review by proudofsimin — COMMENTED** [review]
Hi @vasvi-sood thank you for your work, but I am quite confused that I remembered last time we said that we will remove the section of "create the bot" since we have the installation section in tutorials. Why are we doing it again?

File: README.md [hunk 4]
```diff
@@ -120,14 +120,25 @@ At the same time, we also meet up offline all over the world. Here are some acti
 - [Wechaty Country Wide Developer Conference](https://wechaty.js.org/2018/09/15/country-wide-developer-conference/)
 - .....
 
+### How to Run Locally
```
Comment (Rohitesh-Kumar-Jain) [inline README.md:123]: Is this section finally in the documentation?

**Comment by Rohitesh-Kumar-Jain** [comment]
Hi, 

I followed these steps but in the end maybe I am doing something wrong, I am unable to get ```dong``` for a ```ding```

**Comment by Rohitesh-Kumar-Jain** [comment]
What is to be done after scanning the code

**Comment by vasvi-sood** [comment]
@Rohitesh-Kumar-Jain  You need to scan the code with whatsapp web, then ask a friend to send you a ding message. Please ensure ding is lowercase.

**Comment by Rohitesh-Kumar-Jain** [comment]
> You need to scan the code with whatsapp web

Do I need to keep WhatsApp web open while scanning on the phone?

**Review by Rohitesh-Kumar-Jain — CHANGES_REQUESTED** [review]
Improper PR, pls see my comments on #940 

How to guides team your documentation quality is good but your team is having some serious issues with PRs and git 😕

**Comment by Rohitesh-Kumar-Jain** [comment]
Try to fix the issues yourself first, and if you aren't able to fix these issues I will demonstrate how to work on two features simultaneously at the next meeting which will be nothing but the implementation of these steps https://gist.github.com/Rohitesh-Kumar-Jain/54e9f93bb6fad32ea2385016d1e2858d

One way to fix this error is :  

1) Switch to master

`git switch master`

2) make a new branch for this PR

`git switch -c branch-name`

3) copy paste work related to this PR (related to only this PR).

4) commit changes from that branch 

5) push changes 

` git push origin branch-name`

6) make a new PR, and close this one with reference to this one

7) congrats one PR solved 

8) now, switch to master

`git switch master`

repeat steps 1-8 for other faulty PRs (all of your PRs are faulty 😅)

Don't wait for mentors to close PRs, close yourself : )

PRs like this are unacceptable and are going to cause merge conflict as improperly rebased.

**Comment by abhishek-iiit** [comment]
Hello @Rohitesh-Kumar-Jain, I did exactly the same way, as you suggested here as well as in your previous comment. But, all the commits that were one in the master branch show too in the created branch. PR #939 and #940 are done exactly in the same manner.

**Comment by Rohitesh-Kumar-Jain** [comment]
>  But, all the commits that were one in the master branch show too in the created branch.

You aren't supposed to commit anything to the master branch, always remember never to commit directly or change anything in your master branch. You just need to keep your master branch in sync with the upstream that's it.

No where in my comments I mentioned you to make commits in your master branch.

**Comment by Rohitesh-Kumar-Jain** [comment]
@abhishek-iiit pls first backup all your documentation, I am having a fear that you might lose all your progress while making changes.

I see your master is 41 commits ahead the upstream, that needs to be corrected first.

**Comment by Rohitesh-Kumar-Jain** [comment]
I am saying again backup all of your progress in some files first, before doing his manovuer 

## fix your current master branch

First of all, checkout to your master branch, and then checkout to a new PR branch based on that.

```
git switch master
git switch -c new-branch
```

Then push the new branch to your fork and create a new PR based on that branch (the PR will be identical to this one with the exception of the branch). When you've done that, we can reset your master branch.

```
git switch master
git fetch upstream master # fetch upstream/master without merging into local master
git reset --hard upstream/master # reset local master to upstream/master's state
git push --force origin master # force-update your fork's master to reflect upstream
```

**Comment by Rohitesh-Kumar-Jain** [comment]
@huan and @lijiarui 

What do you think should be done in this situation ?

**Comment by abhishek-iiit** [comment]
> @abhishek-iiit pls the first backup all your documentation, I am having a fear that you might lose all your progress while making changes.
Took the backup of the documentation. Thank you, for the help. Now, will make PR with suggested changes :)

**Comment by vasvi-sood** [comment]
> Hi @vasvi-sood thank you for your work, but I am quite confused that I remembered last time we said that we will remove the section of "create the bot" since we have the installation section in tutorials. Why are we doing it again?

We thought that instaling wechaty meant the six line code to get started with wechaty. We wanted to create a guide for using wechaty with Whatsapp and Wechat separately because those are the most popular messaging apps. So it helps if developers find a separate section for that.

**Comment by vasvi-sood** [comment]
> > You need to scan the code with whatsapp web
> 
> Do I need to keep WhatsApp web open while scanning on the phone?

No, you don't need to. Can you please share the error you are getting.

File: docusaurus/docs/howto/starter.md (hunk 2 above)
Comment (vasvi-sood) [reply docusaurus/docs/howto/starter.md:5]: > In this section, you will learn how to run the bot on the popular messaging apps, including WhatsApp and Wechat.
That would be better. I'll do the necessary changes

**Comment by Rohitesh-Kumar-Jain** [comment]
> No, you don't need to. Can you please share the error you are getting.

I am not getting any error, after I scan the QR code, even my phone gets registered, but where to go from there? 

I got on 'dong's for 'ding's send to my phone

**Comment by Rohitesh-Kumar-Jain** [comment]
I will suggest you to first focus on your existing PRs, correct them.
And don't forget to backup your work. (I am having a fear that you might lose your work while trying to fix git issues, and that will be a disaster, so pls the first backup and save everything you have documented so far, and then feel free to refactor your PRs)

**Comment by lijiarui** [comment]
> @huan and @lijiarui
>
> What do you think should be done in this situation ?

@Rohitesh-Kumar-Jain It seems this PR commit too many files unrelated to this PR. If @vasvi-sood found it's hard to use git to correct them, a better way maybe close this PR, create a new branch and create a new PR.

**Comment by Rohitesh-Kumar-Jain** [comment]
@vasvi-sood and @abhishek-iiit I have a really simple idea to fix your issues if you are finding git way hard to fix issues.

@abhishek-iiit create a fresh fork (don't delete this fork, as this fork anyways belongs to @vasvi-sood).

And then just follow the instructions I gave you to work on more than 1 features, and raise new PRS, and we will close these PRs.

**Comment by abhishek-iiit** [comment]
Thank you for your guidance and support, we have done the same way to fork the project(doesn't delete the previous one) and will open PR with the new forked repo.

**Comment by shraddhavp** [comment]
@abhishek-iiit  and @vasvi-sood  thanks for the PR it is actually helpful for installing Wechaty locally ,one thing I would like to suggest is ,as all the installation steps   mentioned are easy for Linux users but Windows users place difficulty to run some of the commands like sudo etc., can you please add a section for Windows user too? as it would be even more widely reachable

**Comment by proudofsimin** [comment]
> @abhishek-iiit and @vasvi-sood thanks for the PR it is actually helpful for installing Wechaty locally ,one thing I would like to suggest is ,as all the installation steps mentioned are easy for Linux users but Windows users place difficulty to run some of the commands like sudo etc., can you please add a section for Windows user too? as it would be even more widely reachable

Hi @shraddhavp, 
Thank you for your great suggestion. This PR is accidentally overlap with #905 from the tutorial team, so it's preferred if you could notify in there.

**Pull request closed without merging**